src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,239 @@
+  const { todos, setTodos } = useContext(TodosContext);
+  const { tempTodo, setTempTodo } = useContext(TempTodoContext);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [status, setStatus] = useState('all');
+  const { errorMessage, setErrorMessage } = useContext(ErrorContext);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const refInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    todoService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(Error.LOAD_TODOS);
+      })
+      .finally(() => {
+        setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+      });
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (status) {
+      case Filter.Active:
+        return !todo.completed;
+
+      case Filter.Completed:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
+
+  const completedTodo = todos.some(todo => todo.completed);
+
+  const addTodo = () => {
+    todoService.createTodo({
+      userId: USER_ID,
+      title: todoTitle.trim(),
+      completed: false,
+    })
+      .then(newTodo => {
+        setTodos([...todos, newTodo]);
+        setTodoTitle('');
+        setTempTodo(null);
+      })
+      .catch((error) => {
+        setErrorMessage(Error.ADD_TODO);
+        setTempTodo(null);
+        setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+        throw error;
+      });
+  };
+
+  const handleOnSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!tempTodo) {
+      setTempTodo({
+        id: 0,
+        userId: USER_ID,
+        title: todoTitle.trim(),
+        completed: false,
+      });
+    }
+
+    if (!todoTitle.trim()) {
+      setErrorMessage(Error.EMPTY_TITLE);
+      setTempTodo(null);
+
+      setTimeout(() => {
+        setErrorMessage('');
+      }, ERROR_DELAY);
+    } else {
+      addTodo();
+    }
+  };
+
+  useEffect(() => {
+    if (refInput.current) {
+      refInput.current.focus();
+    }
+  }, [tempTodo, todos.length]);
+
+  const count = todos.filter(todo => !todo.completed).length;
+  const isCompleted = todos.every(todo => todo.completed);
+
+  const handleToggleAll = () => {
+    const updateTodos = todos
+      .map(todo => ({ ...todo, completed: !isCompleted }));
+
+    const updatePromises = todos.filter(t => t.completed === isCompleted);
+
+    setLoadingIds(updatePromises.map(t => t.id));
+
+    updatePromises.forEach(todo => {
+      todoService.updateTodo(todo)
+        .then(() => {
+          setTodos(updateTodos);
+          setLoadingIds([]);
+        })
+        .catch(() => {
+          setErrorMessage(Error.UPDATE_TODO);
+          setLoadingIds([]);
+
+          setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+        });
+    });
+  };
+
+  const clearAllCompleted = () => {
+    const updatedTodos = todos.filter(t => !t.completed);
+    const Ids = todos.filter(t => t.completed).map(t => t.id);
```

Comment:
```suggestion
    const todoIds = todos.filter(t => t.completed).map(t => t.id);
```
camel case for common variables, not PascalCase